### PR TITLE
ui: deprecate old UI and move to legacy to be served at /client/legacy

### PR DESCRIPTION
**Pull request by rhtyd**
As notified in the 4.14.0.0 release notes, this deprecates the legacy UI
and moves it under the ui/legacy directory which will be served at
/client/legacy URL path. This will also ensure that users upgrading or
installing 4.15.0.0 will get the new UI by default. This will also make
it simpler to remove the old UI for future 4.16/master releases.

The removal notice has been added here: https://github.com/apache/cloudstack-documentation/blob/master/source/releasenotes/about.rst#primate-ga-and-legacy-ui-deprecation-and-removal-notice

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [x] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by rhtyd** [comment]
This is strictly UI only change, screenshots:
Screenshot from 2020-12-04 17-28-58
Screenshot from 2020-12-04 17-29-19

I'll add details on what we're doing here on the on-going email thread on dev@ ML later today.

**Comment by rhtyd** [comment]
Basic sanitation check:
@blueorangutan package

File: ui/index.html [hunk 1]
```diff
@@ -16,1978 +16,14 @@
...
-
-        <!-- Plugin/module API -->
-        <script type="text/javascript" src="scripts/ui-custom/pluginListing.js"></script>
-        <script type="text/javascript" src="plugins/plugins.js"></script>
-        <script type="text/javascript" src="modules/modules.js"></script>
-        <script type="text/javascript" src="scripts/plugins.js"></script>
-
-        <!-- Load this script after all scripts have executed to populate data -->
-        <script type="text/javascript" src="scripts/postLoad.js"></script>
+        <p>The legacy UI has been deprecated in this version as notified in the <a href="http://docs.cloudstack.apache.org/en/4.14.0.0/releasenotes/about.html#new-user-interface-depreciation-notice-of-existing-ui">previous release</a>. The legacy UI will be <a href="http://docs.cloudstack.apache.org/en/4.15.0.0/releasenotes/about.html#primate-ga-and-legacy-ui-deprecation-and-removal-notice">removed in the next release</a>.<br/>To access the legacy UI <a href="legacy">click here</a>.</p>
```
Comment (DaanHoogland) [inline ui/index.html:27]: I wouldn't even make it a link, justr the instruction "add '/lagacy' to the url in your browser"

File: ui/legacy/index.html [hunk 2]
```diff
@@ -0,0 +1,1993 @@
...
+        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+        <title></title>
+        <link rel="shortcut icon" href="images/cloud.ico" />
+        <link type="text/css" rel="stylesheet" href="lib/jquery-ui/css/jquery-ui.css" />
+        <link type="text/css" rel="stylesheet" href="lib/reset.css"/>
+        <link type="text/css" rel="stylesheet" href="css/cloudstack3.css" />
+        <link type="text/css" rel="stylesheet" href="css/custom.css" />
+    </head>
+    <body>
+        <!-- CloudStack widget content -->
```
Comment (DaanHoogland) [inline ui/legacy/index.html:32]: should we add a big deprecation banner here as well?

**Review by wido — APPROVED** [review]
LGTM

Nice work all!

File: ui/index.html (hunk 1 above)
Comment (rhtyd) [reply ui/index.html:27]: Actually most users won't see this as the new UI (Primate) will get installed over the folder overwriting this index.html with its own.

File: ui/legacy/index.html (hunk 2 above)
Comment (rhtyd) [reply ui/legacy/index.html:32]: I think we could, let me see if I can put something together.

**Comment by rhtyd** [comment]
Thanks @wido 

@DaanHoogland I've added the banner, can you review:
Screenshot from 2020-12-04 23-13-55

**Comment by rhtyd** [comment]
@blueorangutan package

**Review by GabrielBrascher — APPROVED** [review]
Thanks for the changes @rhtyd!

LGTM

**Comment by rhtyd** [comment]
Did several tests, UI works OK after moving to legacy/ path. I'll merge this to test the bundling of primate build with 4.15/master pkgs next. Merging based on 2 lgtms and manual testing.

**Pull request merged**